### PR TITLE
Bundle for browser on build

### LINKED ISSUE #82: Define strategy to bundle browser files
**Is your feature request related to a problem? Please describe.**
Following up on #64, we should define a strategy to make the built JavaScript file from Clio source ready to run in the browser.

**Describe the solution you'd like**
`clio build` script should do some steps:
1 - Compile the Clio source into JS code (this is being done already);
2 - Install dependencies on the bundle directory (there will be a package.json there);
3 - Call an NPM script (declared in the package.json) to bundle the JavaScript files;

`clio build` should take a flag `--no-bundle` that should avoid step three from happening.

Just like `clio build`, `clio run` should cover all the 3 steps describe above, and execute the 4th step to make the compiled source available through a static HTTP server.

**Describe alternatives you've considered**
Inside the build directory (conventioned as `build/browser`), we should have a package.json with two scripts to bundle and host. We are considering using *parcel* to bundle because it is a zero-configuration bundler. For the HTTP I suggest `http-server`.

**Additional context**

### PR DESCRIPTION
**Pull request by andreyluiz**
Closes  #82.

### CODE REVIEW

File: cli/commands/build.js [hunk 1]
```diff
@@ -28,38 +31,40 @@ const mkdir = directory => {
 function getDestinationFromConfig(source, target) {
   const config = packageConfig.getPackageConfig();
   const buildConfig = config.build;
+  const buildDirectory = buildConfig.directory;
 
-  if (!buildConfig) {
+  if (!buildDirectory) {
     throw new Error(
-      'No build configuration has been found. It is a "[build]" section on you "clio.toml" file.'
+      'The build directory is missing on your "clio.toml".\n\nExample:\n\n[build]\ndirectory = "build"\n'
```
Comment (garritfra) [inline cli/commands/build.js]: I notice that you use single quotes. Are you using Prettier locally? Or is it because you have nested quotes?

Comment (andreyluiz) [reply cli/commands/build.js]: It's because there are nested quotes on`"clio.toml`". I run prettier every time I save. :)

Comment (garritfra) [reply cli/commands/build.js]: Ahh okay, perfect! :)

File: cli/commands/build.js [hunk 2]
```diff
@@ -30,42 +34,60 @@ const mkdir = directory => {
...
+  if (!config) {
+    throw new Error('You must pass the location of the "clio.toml" file.');
+  }
+  const buildConfig = config.build;
 
-  if (!buildDirectory) {
+  if (!buildConfig) {
     throw new Error(
-      `The build directory is missing on your "${CONFIGFILE_NAME}".\n\nExample:\n\n[build]\ndirectory = "build"\n`
+      `No build configuration has been found. It is a "[build]" section on you "${CONFIGFILE_NAME}" file.`
```
Comment (garritfra) [inline cli/commands/build.js]: Typo: _you_ should be _your_.
Suggestion: Change `It is a "[build]" section on you "${CONFIGFILE_NAME}" file.` to `Please add a [build] section to your "${CONFIGFILE_NAME}" file.`

Comment (andreyluiz) [reply cli/commands/build.js]: Fixed on https://github.com/clio-lang/clio/pull/98/commits/f187fab97d68591dde94202f3c56b9bc30b3b9d5.

File: cli/lib/platforms.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    await bundler.bundle();
+  },
+  async run(path) {
+    const bundler = await setupParcel(path, { watch: true });
+    await bundler.serve();
+  }
+};
+
+const node = {
+  async build() {},
```
Comment (garritfra) [inline cli/lib/platforms.js]: This looks like the node target does not have to be built at all🤔 Maybe we can move the build logic in here? Possibly, we can have a `generic` build target that just outputs js, which will then be used by the node and the web target.

**Comment by pouya-eghbali** [comment]
@andreyluiz #121 [possibly] breaks this. Some changes are required in the web template (`main.clio.js` is now in `build/` instead of `build/src`)

**Comment by andreyluiz** [comment]
@garritfra The build you reported is fixed.

**Comment by andreyluiz** [comment]
@pouya-eghbali Yes, there are some conflicts. I am trying to rebase.

**Comment by garritfra** [comment]
@andreyluiz coverage is fixed

**Pull request merged**